lib/Autocomplete.js

```diff
@@ -254,13 +264,17 @@ class Autocomplete extends React.Component {
   static keyDownHandlers = {
     ArrowDown(event) {
       event.preventDefault()
-      const itemsLength = this.getFilteredItems().length
-      if (!itemsLength) return
+      let items = this.getFilteredItems()
+      if (!items.length) return
+
       const { highlightedIndex } = this.state
-      const index = (
-        highlightedIndex === null ||
-        highlightedIndex === itemsLength - 1
-      ) ?  0 : highlightedIndex + 1
+      let index = highlightedIndex
+
+      for(let i = 0; i < items.length ; i++) {
+        index = (index === null || index === items.length - 1) ?  0 : index + 1
```

Comment:
The `index === null` check only needs to be done once, so it can be moved to where `index` is declared and initialized, which will allow you to remove the branching entirely: `index = (index + 1) % items.length`.